### PR TITLE
globals: Add missing isEqual() function

### PR DESCRIPTION
**Pull request by Turbo87**
Resolves https://github.com/ember-cli/ember-rfc176-data/issues/5

### CODE REVIEW

**Comment by rwjblue** [comment]
Why do we need this? What is wrong with `===`?

**Comment by Turbo87** [comment]
from https://emberjs.com/internal/#method_isEqual:

> isEqual is a more specific comparison than a triple equal comparison. It will call the isEqual instance method on the objects being compared, allowing finer control over when objects should be considered equal to each other.

It's public API so unless it's deprecated API we should have an import path for it IMHO

**Comment by rwjblue** [comment]
> It's public API so unless it's deprecated API we should have an import path for it IMHO

Please see the discussion in the RFC. The globals API is definitely public, and is not deprecated at this time, but that definitely does **not** mean that we have to include it in the new go forward modules API.

**Comment by Turbo87** [comment]
let me rephrase, how is `isEqual()` less important than e.g. `isPresent()`?

**Comment by rwjblue** [comment]
Yep, I totally understand your point.  I'm not sure of the answer.  The RFC sadly did not incorporate all of Ember's public global API's so we are somewhat left in the lurch regarding what to do with the ones that are missing.

After my last comment, I re-read the RFC prose and it seems that we _should_ be including all current public globals APIs.

> Because our goal is to eliminate the Ember global object, any public classes, functions or properties that currently exist on the global need an equivalent module that can be imported.

I'm just not sure exactly how we are supposed to determine where they go.

**Comment by Turbo87** [comment]
> I'm just not sure exactly how we are supposed to determine where they go.

Anarchy?! :trollface:

**Comment by rwjblue** [comment]
Anarchy rocks
